### PR TITLE
Feat/privacy settings

### PR DESCRIPTION
**Pull request by fsansalvadore**
## What kind of change does this PR introduce?

- Added modal to manage privacy preferences in www and docs.
- Now when user accepts telemetry from ConsentToast, a telemetry request is sent so we collect that page visit

Screenshot 2023-10-04 at 17 49 45

### CODE REVIEW

**Comment by pamelachia** [comment]
LGTM!!!

**Review by pamelachia — APPROVED** [review]

File: apps/docs/components/Navigation/Footer.tsx [hunk 1]
```diff
@@ -44,11 +44,17 @@ const Footer = () => (
...
         <span className="text-xs text-scale-900">—</span>
-        {secondaryLinks.map((item) => (
-          <Link href={item.url} key={item.url}>
-            <a className="text-xs text-scale-800 hover:underline">{item.title}</a>
-          </Link>
-        ))}
+        {secondaryLinks.map(({ component: Component, ...item }) =>
+          item.url ? (
+            <Link href={item.url} key={item.url}>
+              <a className="text-xs text-scale-800 hover:underline">{item.title}</a>
```
Comment (MildTomato) [inline apps/docs/components/Navigation/Footer.tsx:50]: text-foreground-lighter

File: apps/docs/components/Navigation/Footer.tsx [hunk 2]
```diff
@@ -44,11 +44,17 @@ const Footer = () => (
...
+            </Link>
+          ) : (
+            Component && (
+              <Component className="text-xs text-scale-800 hover:underline">{item.title}</Component>
```
Comment (MildTomato) [inline apps/docs/components/Navigation/Footer.tsx:54]: same

File: apps/www/pages/privacy.mdx [hunk 3]
```diff
@@ -331,9 +332,9 @@ If you would like to find out more about cookies and other similar technologies,
...
 
-**Cookiesless analytics**
+#### **Cookiesless analytics**
 
-We also use cookieless technologies to collect information to assess how our Service is used. When you access a website or service on the internet, your device and your browser send certain information to the provider of that website or service in order to access that website or service. We collect, aggregate and analyze this information to identify errors in and improvements to Service. We will only do so, however, only if you give us your consent to do so.
+We also use cookieless technologies to collect information to assess how our Service is used. When you access a website or service on the internet, your device and your browser send certain information to the provider of that website or service in order to access that website or service. We collect, aggregate and analyze this information to identify errors in and improvements to Service. We will only do so, however, only if you give us your consent to do so. <PrivacySettings className="underline">Manage preferences</PrivacySettings>
```
Comment (MildTomato) [inline apps/www/pages/privacy.mdx:337]: alternative could be using queryParam for triggering opening or closing settings. 

then its just something like `<a href="?privacySettings=true">`

**Review by MildTomato — CHANGES_REQUESTED** [review]

File: apps/www/pages/privacy.mdx (hunk 3 above)
Comment (MildTomato) [reply apps/www/pages/privacy.mdx:337]: dont worry about it though

**Comment by MildTomato** [comment]
There's also a minor bug in which before dismissing the opt in popover, you can open privacy settings, opt in, and the opt in popover is still visible. 

Not a blocker to prod though.

File: apps/docs/components/Navigation/Footer.tsx (hunk 1 above)
Comment (fsansalvadore) [reply apps/docs/components/Navigation/Footer.tsx]: Ok, migrated every occurrence of scale colors in www and docs footers (borders, text etc) @MildTomato

File: apps/docs/components/Navigation/Footer.tsx [hunk 4]
```diff
@@ -37,18 +37,24 @@ const Footer = () => (
         ))}
       </ul>
     </div>
-    <hr className="border-scale-400 my-6"></hr>
+    <hr className="border-border my-6"></hr>
```
Comment (MildTomato) [inline apps/docs/components/Navigation/Footer.tsx:40]: can just be `border` if you want.

**Review by MildTomato — CHANGES_REQUESTED** [review]

File: apps/docs/components/Navigation/Footer.tsx (hunk 4 above)
Comment (fsansalvadore) [reply apps/docs/components/Navigation/Footer.tsx:40]: I believe just `border` competes with actually setting all borders.

File: apps/www/components/Footer/index.tsx [hunk 5]
```diff
@@ -82,7 +82,7 @@ const Footer = (props: Props) => {
 
               <a
                 href="https://github.com/supabase"
-                className="text-scale-900 hover:text-scale-1200 transition"
+                className="text-lighter hover:text-strong transition"
```
[the remark this answers is not in the document]
Comment (fsansalvadore) [reply apps/www/components/Footer/index.tsx:85]: Why not shorthand for text? For clarity?

**Comment by fsansalvadore** [comment]
> There's also a minor bug in which before dismissing the opt in popover, you can open privacy settings, opt in, and the opt in popover is still visible.

This is now fixed @MildTomato

File: apps/www/components/Footer/index.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (fsansalvadore) [reply apps/www/components/Footer/index.tsx]: hover:text-foreground done ✅

**Review by MildTomato — APPROVED** [review]

**Pull request merged**